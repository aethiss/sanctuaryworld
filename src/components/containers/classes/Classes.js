import React from 'react'

// Material-ui
import ButtonGroup from '@material-ui/core/ButtonGroup'
import DiabloButton from '../../commons/buttons/diabloButton/DiabloButton'

const styles = {
  classesContainer: {
    position: 'relative',
  },
  classesSelection: {
    margin: 0,
    position: 'absolute',
    top: '20%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    zIndex: 1,
    textAlign: 'center'
  },
  classesButtons: {
    marginLeft: '10px',
    padding: '10px',
  }
}

const Classes = () => {
  return (
    <div style={styles.classesContainer}>
      <video key="classes-loop-video" autoPlay loop muted width="100%">
        <source src="https://d3a5h34gwy5glx.cloudfront.net/assets/media/classes-bg-loop-ultra.40931d6dfc85ffaca3d9010051b0d648.mp4" />
      </video>
      <div style={styles.classesSelection}>
        <ButtonGroup aria-label="outlined primary button group">
          <DiabloButton text="Barbarian" isDisabled enableOnOver anchorLink='/class/barbarian' />
          <DiabloButton text="Sorceress" isDisabled enableOnOver anchorLink='/class/sorcerer' />
          <DiabloButton text="Druid" isDisabled enableOnOver anchorLink='/class/druid' />
        </ButtonGroup>
      </div>
    </div>
  )
}

export default Classes
