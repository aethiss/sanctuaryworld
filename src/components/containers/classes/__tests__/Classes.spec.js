import React from 'react'
import TestRenderer from 'react-test-renderer'

// Component
import Classes from '../Classes'

describe('Classes', () => {
  describe('Render', () => {
    it('[Classes]: Default Render Classes', () => {
      const testRenderer = TestRenderer.create(<Classes />)
      expect(testRenderer).toMatchSnapshot()
    })
  })
})
