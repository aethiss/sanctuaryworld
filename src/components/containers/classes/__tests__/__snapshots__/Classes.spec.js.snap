// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Classes Render [Classes]: Default Render Classes 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <video
    autoPlay={true}
    loop={true}
    muted={true}
    width="100%"
  >
    <source
      src="https://d3a5h34gwy5glx.cloudfront.net/assets/media/classes-bg-loop-ultra.40931d6dfc85ffaca3d9010051b0d648.mp4"
    />
  </video>
  <div
    style={
      Object {
        "left": "50%",
        "margin": 0,
        "position": "absolute",
        "textAlign": "center",
        "top": "20%",
        "transform": "translate(-50%, -50%)",
        "zIndex": 1,
      }
    }
  >
    <div
      aria-label="outlined primary button group"
      className="MuiButtonGroup-root"
      role="group"
    >
      <a
        href="/class/barbarian"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <button
          className="makeStyles-base-1"
          onClick={[Function]}
        >
          Barbarian
        </button>
      </a>
      <a
        href="/class/sorcerer"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <button
          className="makeStyles-base-2"
          onClick={[Function]}
        >
          Sorceress
        </button>
      </a>
      <a
        href="/class/druid"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <button
          className="makeStyles-base-3"
          onClick={[Function]}
        >
          Druid
        </button>
      </a>
    </div>
  </div>
</div>
`;
